sentry-android-core/src/main/java/io/sentry/android/core/util/ConnectivityChecker.java

```diff
@@ -16,98 +17,163 @@
-        return null;
-      }
-      Network activeNetwork = connectivityManager.getActiveNetwork();
+  @SuppressLint({"ObsoleteSdkInt", "MissingPermission", "NewApi"})
+  public static @Nullable String getConnectionType(
+      final @NotNull Context context,
+      final @NotNull ILogger logger,
+      final @NotNull IBuildInfoProvider buildInfoProvider) {
+    final ConnectivityManager connectivityManager = getConnectivityManager(context, logger);
+    if (connectivityManager == null) {
+      return null;
+    }
+    if (!Permissions.hasPermission(context, Manifest.permission.ACCESS_NETWORK_STATE)) {
+      logger.log(SentryLevel.INFO, "No permission (ACCESS_NETWORK_STATE) to check network status.");
+      return null;
+    }
+
+    boolean ethernet = false;
+    boolean wifi = false;
+    boolean cellular = false;
+
+    if (buildInfoProvider.getSdkInfoVersion() >= Build.VERSION_CODES.M) {
+      final Network activeNetwork = connectivityManager.getActiveNetwork();
       if (activeNetwork == null) {
         logger.log(SentryLevel.INFO, "Network is null and cannot check network status");
         return null;
       }
-      NetworkCapabilities networkCapabilities =
+      final NetworkCapabilities networkCapabilities =
           connectivityManager.getNetworkCapabilities(activeNetwork);
       if (networkCapabilities == null) {
         logger.log(SentryLevel.INFO, "NetworkCapabilities is null and cannot check network type");
         return null;
       }
-      if (networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)) {
-        return "wifi";
-      }
       if (networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)) {
-        return "ethernet";
+        ethernet = true;
+      }
+      if (networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)) {
+        wifi = true;
       }
       if (networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
-        return "cellular";
+        cellular = true;
+      }
+    } else {
+      // ideally using connectivityManager.getAllNetworks(), but its >= Android L only
+
+      // for some reason linting didn't allow a single @SuppressWarnings("deprecation") on method
+      // signature
+      @SuppressWarnings("deprecation")
+      final android.net.NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
+
+      if (activeNetworkInfo == null) {
+        logger.log(SentryLevel.INFO, "NetworkInfo is null, there's no active network.");
+        return null;
+      }
+
+      @SuppressWarnings("deprecation")
+      final int type = activeNetworkInfo.getType();
+
+      @SuppressWarnings("deprecation")
+      final int TYPE_ETHERNET = ConnectivityManager.TYPE_ETHERNET;
+
+      @SuppressWarnings("deprecation")
+      final int TYPE_WIFI = ConnectivityManager.TYPE_WIFI;
+
+      @SuppressWarnings("deprecation")
+      final int TYPE_MOBILE = ConnectivityManager.TYPE_MOBILE;
```

Comment:
`l`: What about using them in the switch statement directly?

Reply:
https://github.com/getsentry/sentry-android/pull/420#discussion_r427911518

that was my first try but cant use `@SuppressWarnings("deprecation")` on an enum